packages/react/src/auto/shadcn/inputs/ShadcnAutoDateTimePicker.tsx

```diff
@@ -68,7 +68,12 @@ export const makeShadcnAutoDateTimePicker = ({
 
     const localTz = Intl.DateTimeFormat().resolvedOptions().timeZone;
     const localTime = useMemo(() => {
-      return value ? value : isValidDate(new Date(fieldProps.value)) ? new Date(fieldProps.value) : undefined;
+      if (value) return value;
+
+      if (fieldProps.value == null) return undefined; // Prevents null from becoming 1970
+
+      const date = new Date(fieldProps.value);
+      return isValidDate(date) ? date : undefined;
     }, [value, fieldProps.value]);
```

Comment:
Since we have to make this change in two places, I think it'd be good to refactor this out into a common utility so that the next design system just gets these smarts by default!